internal/providers/msentraid/himmelblau.go

```diff
@@ -34,23 +35,49 @@ var (
 	ErrDeviceDisabled = fmt.Errorf("device is disabled in Microsoft Entra ID, please contact your administrator")
 )
 
-func init() {
-	var err *C.MSAL_ERROR
+// TokenAcquisitionError is returned when an error occurs while acquiring a token via libhimmelblau.
+type TokenAcquisitionError struct {
+	msg string
+}
 
-	err = C.set_global_tracing_level(C.TRACE)
-	if err != nil {
-		panic(fmt.Sprintf("failed to set global tracing level: %v", C.GoString(err.msg)))
-	}
+func (e TokenAcquisitionError) Error() string {
+	return e.msg
+}
 
-	// An optional TPM Transmission Interface. If this parameter is NULL, a soft TPM is initialized.
-	var tctiName *C.char
-	err = C.tpm_init(tctiName, &tpm)
-	if err != nil {
-		panic(fmt.Sprintf("failed to initialize TPM: %v", C.GoString(err.msg)))
-	}
+func ensureTPMInitialized() (err error) {
+	tpmInitOnce.Do(func() {
+		filter := "warn"
+		logLevel := log.GetLevel()
+		if logLevel <= log.DebugLevel {
+			log.Debug(context.Background(), "Setting libhimmelblau tracing level to DEBUG")
+			filter += ",himmelblau=debug"
+		} else if logLevel <= log.InfoLevel {
+			filter += ",himmelblau=info"
+		}
+
+		msalErr := C.set_module_tracing_filter(C.CString(filter))
```

Comment:
Does this allows us to drop the `RUST_LOG` thingy?

Reply:
indeed, I forgot to drop `RUST_LOG`. did that now. thanks!